config/v1/types_infrastructure.go

```diff
@@ -1277,6 +1307,45 @@ type VSpherePlatformTopology struct {
 	Template string `json:"template,omitempty"`
 }
 
+// VSphereFailureDomainAffinity contains the vCenter cluster vm-host group (virtual machine and host types)
+// and the vm-host affinity rule that together creates a affinity configuration for vm-host based zonal.
+// This configuration within vCenter creates the required association between a failure domain, virtual machines
+// and ESXi hosts to create a vm-host based zone.
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'HostGroup' ?  has(self.hostGroup) : !has(self.hostGroup)",message="hostGroup is required when type is HostGroup, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'HostGroup' ?  has(self.vmGroup) : !has(self.vmGroup)",message="vmGroup is required when type is HostGroup, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'HostGroup' ?  has(self.vmHostRule) : !has(self.vmHostRule)",message="vmHostRule is required when type is HostGroup, and forbidden otherwise"
+// +union
+type VSphereFailureDomainAffinity struct {
```

Comment:
I'm not sure I'm following the validation requirements of these fields, for vmGroup, hostGroup and vmHostRule, could you fill the following table

| Type | HostGroup | ComputeCluster | DataCenter |
| --- | --- | --- | --- |
| vmGroup | | | |
| hostGroup | | | |
| vmHostRule | | | |

With values `R` for required, `O` for optional, or `N/A` if the field has no effect for that particular `Type` value

Reply:
> I'm not sure I'm following the validation requirements of these fields, for vmGroup, hostGroup and vmHostRule, could you fill the following table

|Type|HostGroup| ComputeCluster |DataCenter
|---|---|---|---|
|vmGroup|R|N/A|N/A|
hostGroup |R|N/A|N/A|
vmHostRule|R|N/A|N/A|

> With values `R` for required, `O` for optional, or `N/A` if the field has no effect for that particular `Type` value